### DISCUSSION TITLE
shasum is always empty for any package metadata received from packagist.org

### DISCUSSION THREAD

**Discussion by nareshdevasani** [composer/composer#11489] [Q&A / Support]
Hello, I need help in getting `shasum` for any given php package. I wanted to validate the integrity of the package.
Below is the API used but `shasum` is always empty.
`https://repo.packagist.org/p2/guzzlehttp/guzzle.json`
But the response does not contain `shasum`. And I did not find it for any package I tried.

```
                "dist": {
                    "url": "https://api.github.com/repos/guzzle/guzzle/zipball/fb7566caccf22d74d1ab270de3551f72a58399f5",
                    "type": "zip",
                    "shasum": "",
                    "reference": "fb7566caccf22d74d1ab270de3551f72a58399f5"
                },
```

**Question:** Is `shasum` right field for getting checksum of any given PHP package?
What is the reference value above and that is same in the URL as well.

**Comment by stof** [comment]
This is because neither Github nor Gitlabs provide a stable shasum for their dist archives. And most packages have archives coming from Github or Gitlab

**Comment by fredden** [comment]
You may like to investigate https://github.com/sansecio/composer-integrity-plugin as a way to validate integrity of packages.
